metaserver/src/main/scala/scala/meta/languageserver/Ctags.scala

```diff
@@ -0,0 +1,238 @@
+package scala.meta.languageserver
+
+import java.io.IOException
+import java.nio.charset.StandardCharsets
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.zip.ZipInputStream
+import scala.collection.GenSeq
+import scala.collection.parallel.mutable.ParArray
+import scala.meta.Defn
+import scala.meta.Defn
+import scala.meta.Name
+import scala.meta.Pat
+import scala.meta.Pkg
+import scala.meta.Source
+import scala.meta.Template
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.Type
+import scala.meta.parsers.ParseException
+import scala.meta.transversers.Traverser
+import scala.util.control.NonFatal
+import com.typesafe.scalalogging.LazyLogging
+import org.langmeta.inputs.Input
+import org.langmeta.inputs.Position
+import org.langmeta.internal.io.FileIO
+import org.langmeta.internal.io.PathIO
+import org.langmeta.io.AbsolutePath
+import org.langmeta.io.Fragment
+import org.langmeta.io.RelativePath
+import org.langmeta.semanticdb._
+
+/**
+ * Syntactically build a semanticdb containing only global symbol definition.
+ *
+ * The purpose of this module is to provide "Go to definition" from
+ * project sources to dependency sources without indexing classfiles or
+ * requiring dependencies to publish semanticdbs alongside their artifacts.
+ *
+ * One other use-case for this module is to implement "Workspace symbol provider"
+ * without any build-tool or compiler integration. Essentially, ctags.
+ */
+object Ctags extends LazyLogging {
+
+  /**
+   * Build an index from a classpath of -sources.jar
+   *
+   * @param shouldIndex An optional filter to skip particular relative filenames.
+   * @param inParallel If true, use parallel collection to index using all
+   *                   available CPU power. If false, uses single-threaded
+   *                   collection.
+   * @param callback A callback that is called as soon as a document has been
+   *                 indexed.
+   */
+  def index(
+      classpath: List[AbsolutePath],
+      shouldIndex: RelativePath => Boolean = _ => true,
+      inParallel: Boolean = true
+  )(callback: Document => Unit): Unit = {
+    val fragments = allClasspathFragments(classpath, inParallel)
+    fragments.foreach { fragment =>
+      try {
+        if (PathIO.extension(fragment.name.toNIO) == "scala" &&
+          shouldIndex(fragment.name)) {
+          callback(index(fragment))
+        }
+      } catch {
+        case _: ParseException => // nothing
+        case NonFatal(e) =>
+          logger.error(e.getMessage, e)
+      }
+    }
+  }
+
+  /** Build a Database for a single source file. */
+  def index(filename: String, contents: String): Document = {
+    val input = Input.VirtualFile(filename, contents)
+    val tree = {
+      import scala.meta._
```

Comment:
Lines 13-24 import various bits of scala.meta, then here you import all.  Why?  Is the style no wildcards allowed at top of file?